testsuite/httpx/__init__.py

```diff
@@ -30,27 +30,27 @@ def __init__(self, retry_codes, response=None, error=None):
 
     def should_backoff(self):
         """True, if the Result can be considered an instability and should be retried"""
-        return self.has_dns_error() or (not self.has_error() and self.status_code in self.retry_codes)
+        return self.has_dns_error() or (self.error is None and self.status_code in self.retry_codes)
 
-    def has_error(self):
-        """True, if the request failed and an error was returned"""
-        return self.error is not None
+    def has_error(self, error_msg: str) -> bool:
+        """True, if the request failed and an error with message was returned"""
+        return self.error is not None and len(self.error.args) > 0 and any(error_msg in arg for arg in self.error.args)
 
     def has_dns_error(self):
         """True, if the result failed due to DNS failure"""
-        return (
-            self.has_error()
-            and len(self.error.args) > 0
-            and any("Name or service not known" in arg for arg in self.error.args)
-        )
-
-    def has_tls_error(self):
-        """True, if the result failed due to TLS failure"""
-        return (
-            self.has_error()
-            and len(self.error.args) > 0
-            and any("SSL: CERTIFICATE_VERIFY_FAILED" in arg for arg in self.error.args)
-        )
+        return self.has_error("Name or service not known")
+
+    def has_tls_cert_verify_error(self):
```

Comment:
I actually prefer the original solution with pytest.raises, as this hides the actual error we are trying to verify. The reason for has_dns_error and has_tls_error to catch any error occurring when we do not care about the specifics

Reply:
I don't think that by calling a method named `has_..._error` we are hiding this error. I see this as an abstraction for error catching that can be used in other tls tests.

Reply:
Another solution can be to have only the `has_error` method and parse the expected error message to it as a parameter